pandas/_libs/groupby_helper.pxi.in

```diff
@@ -954,3 +956,60 @@ def group_shift_indexer(int64_t[:] out, int64_t[:] labels,
                     out[ii] = -1
 
                 label_indexer[lab, idxer_slot] = ii
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def group_fillna_indexer(ndarray[int64_t] out, ndarray[int64_t] labels,
+                         ndarray[uint8_t] mask, **kwargs):
+    """Indexes how to fill values forwards or backwards within a group
+
+    Parameters
+    ----------
+    out : array of int64_t values which this method will write its results to
+        Missing values will be written to with a value of -1
+    labels : array containing unique label for each group, with its ordering
+        matching up to the corresponding record in `values`
+    mask : array of int64_t values where a 1 indicates a missing value
+    direction : {'ffill', 'bfill'}
+        Direction for fill to be applied (forwards or backwards, respectively)
+    limit : Consecutive values to fill before stopping, or -1 for no limit
+
+    Notes
+    -----
+    This method modifies the `out` parameter rather than returning an object
+    """
+    cdef:
+        Py_ssize_t i, N
+        ndarray[int64_t] sorted_labels
+        int64_t limit, idx, curr_fill_idx=-1, filled_vals=0
+
+    direction = kwargs['direction']
```

Comment:
same again why not simply list them in the signature? (you can have **kwargs if you want to *ignore* other ones)

Reply:
I didn't want to define a default value for something like `periods` in both the Python and Cython layers, leaving the former to set that default and be responsible for passing to Cython. 

It's a pretty trivial change so will re-push with your changes reflected soon

Reply:
Think I've been looking at this change too long...planning on making these required in the Cython signature and using the existing defaults in the Python layer to populate those via `**kwargs`. If we are not on the same page let me know

Reply:
no that sounds good